web/components/contract/contracts-table.tsx

```diff
@@ -0,0 +1,284 @@
+      return (
+        <span className={probTextColor}>
+          {getFormattedMappedValue(contract, val)}
+        </span>
+      )
+    }
+    case 'FREE_RESPONSE':
+    case 'MULTIPLE_CHOICE': {
+      return <ContractMinibar contract={contract} />
+    }
+    case 'CERT': {
+      return <span>CERT</span>
+    }
+    case 'QUADRATIC_FUNDING': {
+      return <span>RAD</span>
+    }
+    default:
+      return <span>-</span>
+  }
+}
+
+export function ContractsTable(props: {
+  contracts: Contract[]
+  filter?: filter
+  onContractClick?: (contract: Contract) => void
+  isMobile?: boolean
+  highlightContractIds?: string[]
+  headerClassName?: string
+}) {
+  const {
+    contracts,
+    filter,
+    onContractClick,
+    isMobile,
+    highlightContractIds,
+    headerClassName,
+  } = props
+
+  const router = useRouter()
+  const user = useUser()
+
+  const contractColumns = [
+    {
+      name: 'market',
+      header: 'Market',
+      visible: true,
+      content: (contract: Contract) => (
+        <Row className="gap-4">
+          <Avatar
+            username={contract.creatorUsername}
+            avatarUrl={contract.creatorAvatarUrl}
+            size="xs"
+          />
+          <div className="">{contract.question}</div>
+        </Row>
+      ),
+    },
+    {
+      name: 'prob',
+      header: '%',
+      visible: true,
+      content: (contract: Contract) => (
+        <div className="font-semibold">
+          <ContractStatusLabel contract={contract} />
+        </div>
+      ),
+    },
+    {
+      name: 'action',
+      header: 'Action',
+      visible: !isMobile,
+      content: (contract: Contract) => (
+        <Action contract={contract} user={user} />
+      ),
+    },
+    {
+      name: 'traders',
+      header: 'Traders',
+      visible: !isMobile,
+      content: (contract: Contract) => (
+        <Row className="align-center shrink-0 items-center gap-1">
+          <UserIcon className="h-4 w-4" />
+          {contract.uniqueBettorCount}
+        </Row>
+      ),
+    },
+    {
+      name: 'visibility',
+      header: '',
+      visible: true,
+      content: (contract: Contract) => <Visibility contract={contract} />,
+    },
+  ]
+
+  function ContractRow(props: { contract: Contract }) {
+    const contract = useContract(props.contract.id) ?? props.contract
+    const contractListEntryHighlightClass =
+      'bg-gradient-to-b from-primary-100 via-ink-0 to-ink-0 outline outline-2 outline-primary-400'
+
+    const dataCellClassName = 'py-2 align-top'
+    return (
+      <tr
+        key={contract.id}
+        className={clsx(
+          highlightContractIds?.includes(contract.id)
+            ? contractListEntryHighlightClass
+            : '',
+          (isClosed(contract) && contract.creatorId !== user?.id) ||
+            contract.isResolved
+            ? 'text-ink-500'
+            : '',
+          'hover:bg-primary-50 focus:bg-primary-50 group cursor-pointer'
+        )}
+        onClick={(e) => {
```

Comment:
This needs to be an actual link component. We've done this many times and users always complain. Without a native link, various features break like middle clicking or right clicking to open in a new tab.
I think you can just create the link within the table row: `<tr><Link></Link></tr>`.